item.rb

```diff
@@ -0,0 +1,53 @@
+require 'date'
+
+class Item
+  attr_accessor :publish_date, :archived
+  attr_reader :id, :genres, :authors, :sources, :labels
+
+  def initialize(publish_date, archived: false)
+    @id = Random.rand(1...1000)
+    @publish_date = publish_date
+    @archived = archived
+    @genres = []
+    @authors = []
+    @sources = []
+    @labels = []
+  end
+
+  def add_genre(genre)
+    @genres << genre
+  end
+
+  def add_author(author)
+    @authors << author
+  end
+
+  def add_source(source)
+    @sources << source
+  end
+
+  def add_label(label)
+    @labels << label
+  end
```

Comment:
- [ ] Please consider reviewing your associations: According to the UML, the `Item` class has a `one-to-many` association with `Label`, `Genre`, and `Author`. This means that an `Item` belongs to any of these classes, meaning that it can only have one `Label`, one `Genre`, and one `Author`. However, according to your code, an Item can have multiple instances of these. Please consider fixing this! 🤓 🔧